File: rest-api-ipc2/src/main/java/services/cines/CineAdministradorService.java
package services.cines;

import db.CineAdministradorDB;
import exceptions.EntityNotFoundException;
import models.cines.Cine;
import java.util.List;

public class CineAdministradorService {
    
    private final CineAdministradorDB cineAdministradorDB;
    
    public CineAdministradorService() {
        this.cineAdministradorDB = new CineAdministradorDB();
    }
    
    public List<Cine> getCinesByAdministrador(int idUsuario) throws EntityNotFoundException {
        List<Cine> cines = cineAdministradorDB.getCinesByUsuarioId(idUsuario);
        
        if (cines.isEmpty()) {
            throw new EntityNotFoundException("No se encontraron cines para el usuario administrador con ID: " + idUsuario);
        }
        
        return cines;
    }
}

File: sistema-cines/src/services/cines/cine-administrador.service.ts
import { HttpClient } from "@angular/common/http";
import { Cine } from "../../models/Cines/cine";
import { RestConstants } from "../../shared/restapi/rest-constants";
import { Observable } from "rxjs";
import { Injectable } from "@angular/core";

@Injectable({
  providedIn: 'root'
})
export class CineAdministradorService {
    private restConstants = new RestConstants();

    constructor(private httpClient: HttpClient) { }

    //enviar el id de un usuario administrador de cine para verificar a cuantos cines esta asignado
    public getCinesByAdministrador(idUsuario: number): Observable<Cine[]> {
        return this.httpClient.get<Cine[]>(`${this.restConstants.getApiURL()}cine-admin/${idUsuario}/cines`);
    }
}